src/main/java/duke/storage/Storage.java

```diff
@@ -0,0 +1,119 @@
+package duke.storage;
+
+import duke.exception.DukeException;
+import duke.tasks.Deadline;
+import duke.tasks.Event;
+import duke.tasks.Task;
+import duke.tasks.ToDo;
+
+import java.io.BufferedReader;
+import java.io.BufferedWriter;
+import java.io.File;
+import java.io.FileReader;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.util.LinkedList;
+
+public class Storage {
+
+    private String filepath;
+
+    /**
+     * Initialises a Storage session.
+     *
+     * @param filepath The file to store tasks to.
+     */
+    public Storage(String filepath) {
+        this.filepath = filepath;
+    }
+
+    /**
+     * Reads the input file line by line to add tasks into the TaskList.
+     *
+     * @return The LinkedList containing all loaded tasks.
+     * @throws DukeException In the event that the file cannot be read properly.
+     */
+    public LinkedList<Task> load() throws DukeException {
+        LinkedList<Task> allTasks = new LinkedList<>();
+        try {
+            File f = new File(filepath);
+            if (!f.exists()) {
+                f.createNewFile();
+            }
+            BufferedReader bufferedReader = new BufferedReader(new FileReader(f));
+            String currLine;
+            while ((currLine = bufferedReader.readLine()) != null) {
+                String[] formattedText = currLine.split("\\|");
+                Task t;
+                switch (formattedText[0]) {
+                case "T":
+                    t = new ToDo(formattedText[2]);
+                    break;
+                case "D":
+                    t = new Deadline(formattedText[2], formattedText[3]);
+                    break;
+                case "E":
+                    t = new Event(formattedText[2], formattedText[3]);
+                    break;
+                default:
+                    t = new Task("");
+                    break;
+                }
+                if (formattedText[1].equals("1")) {
+                    t.markAsDone();
+                }
+                allTasks.add(t);
+            }
+        } catch (IOException e) {
+            throw new DukeException(e.getMessage());
+        }
+        return allTasks;
+    }
+
+    /**
+     * Append a new task to the end of the file.
+     *
+     * @param t The task to be added.
+     */
+    public void appendTaskToFile(Task t) {
+        try {
+            BufferedWriter output = new BufferedWriter(new FileWriter(filepath, true));
```

Comment:
You can use the `FileWriter` to write file directly without the need for `BufferedWriter`. See the following example:
```Java
FileWriter writer = new FileWriter(filePath, true); // Note the true here to enable append mode
writer.write("This is one line\n");
writer.flush();
writer.close();
```